sys-libs/libcxx/libcxx-9999.ebuild

```diff
@@ -21,89 +25,99 @@ fi
 
 LICENSE="|| ( UoI-NCSA MIT )"
 SLOT="0"
-if [ "${PV%9999}" = "${PV}" ] ; then
-	KEYWORDS="~amd64 ~mips ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux"
+if [[ ${PV} != 9999 ]] ; then
+	KEYWORDS="~amd64 ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux"
 else
 	KEYWORDS=""
 fi
 IUSE="elibc_glibc elibc_musl +libcxxrt libunwind +static-libs test"
 REQUIRED_USE="libunwind? ( libcxxrt )"
 
-RDEPEND="libcxxrt? ( >=sys-libs/libcxxrt-0.0_p20130725[libunwind?,static-libs?,${MULTILIB_USEDEP}] )
+RDEPEND="libcxxrt? ( sys-libs/libcxxrt[libunwind=,static-libs?,${MULTILIB_USEDEP}] )
 	!libcxxrt? ( >=sys-devel/gcc-4.7:=[cxx] )"
+# llvm-3.9.0 needed because its cmake files installation path changed, which is
+# needed by libcxx
 DEPEND="${RDEPEND}
-	test? ( sys-devel/clang )
-	app-arch/xz-utils"
+	test? ( sys-devel/clang
+		${PYTHON_DEPS} )
+	app-arch/xz-utils
+	>=sys-devel/llvm-3.9.0[${MULTILIB_USEDEP}]"
 
 DOCS=( CREDITS.TXT )
 
+PATCHES=(
+	# Add link flag "-Wl,-z,defs" to avoid underlinking; this is needed in a
+	# out-of-tree build.
+	"${FILESDIR}/${PN}-3.9-cmake-link-flags.patch"
+)
+
 pkg_setup() {
-	if ! use libcxxrt ; then
-		ewarn "You have disabled USE=libcxxrt. This will build ${PN} against"
-		ewarn "libsupc++. Please note that this is not well supported."
-		ewarn "In particular, static linking will not work."
+	use test && python_setup
+
+	if ! use libcxxrt && ! tc-is-gcc ; then
+		eerror "To build ${PN} against libsupc++, you have to use gcc. Other"
+		eerror "compilers are not supported. Please set CC=gcc and CXX=g++"
+		eerror "and try again."
+		die
 	fi
-	if [[ $(gcc-version) < 4.7 ]] && [[ $(tc-getCXX) != *clang++* ]] ; then
-		eerror "${PN} needs to be built with clang++ or gcc-4.7 or later."
-		eerror "Please use gcc-config to switch to gcc-4.7 or later version."
+	if tc-is-gcc && [[ $(gcc-version) < 4.7 ]] ; then
+		eerror "${PN} needs to be built with gcc-4.7 or later (or other"
+		eerror "conformant compilers). Please use gcc-config to switch to"
+		eerror "gcc-4.7 or later version."
 		die
 	fi
 }
 
-src_prepare() {
-	cp -f "${FILESDIR}/Makefile" lib/ || die
-	use elibc_musl && epatch "${FILESDIR}/${P}-musl-support.patch"
-	multilib_copy_sources
+src_unpack() {
+	if use test; then
+		# needed for tests
+		git-r3_fetch "http://llvm.org/git/llvm.git
+			https://github.com/llvm-mirror/llvm.git"
+	fi
+	git-r3_fetch
+
+	if use test; then
+		git-r3_checkout http://llvm.org/git/llvm.git \
+			"${WORKDIR}"/llvm
+	fi
+	git-r3_checkout
```

Comment:
> Some questions:
> - why git fetch and checkout twice?

Because it is using two repositories...

> - why not use a single git clone?

Because that's the eclass API. Fetch and checkout are separate, to account for future phase split.